packages/components/nodes/vectorstores/Postgres/driver/PGVector.ts

```diff
@@ -0,0 +1,119 @@
+                }
+            }
+
+            this._postgresConnectionOptions = {
+                ...additionalConfiguration,
+                host: this.nodeData.inputs?.host as string,
+                port: this.nodeData.inputs?.port as number,
+                user: user,
+                password: password,
+                database: this.nodeData.inputs?.database as string
+            }
+        }
+
+        return this._postgresConnectionOptions
+    }
+
+    async getArgs(): Promise<PGVectorStoreArgs> {
+        return {
+            postgresConnectionOptions: await this.getPostgresConnectionOptions(),
+            tableName: this.getTableName(),
+            columns: {
+                contentColumnName: (this.nodeData.inputs?.contentColumnName || PGVectorDriver.CONTENT_COLUMN_NAME_DEFAULT) as string
+            },
+            distanceStrategy: (this.nodeData.inputs?.distanceStrategy || 'cosine') as DistanceStrategy
+        }
+    }
+
+    async instanciate(metadataFilters?: any) {
+        return this.adaptInstance(await PGVectorStore.initialize(this.getEmbeddings(), await this.getArgs()), metadataFilters)
+    }
+
+    async fromDocuments(documents: Document[]) {
+        const instance = await this.instanciate()
+
+        await instance.addDocuments(documents)
+
+        return this.adaptInstance(instance)
+    }
+
+    protected async adaptInstance(instance: PGVectorStore, metadataFilters?: any): Promise<PGVectorStore> {
+        const { [FLOWISE_CHATID]: chatId, ...pgMetadataFilter } = metadataFilters
+
+        const baseSimilaritySearchVectorWithScoreFn = instance.similaritySearchVectorWithScore.bind(instance)
+
+        instance.similaritySearchVectorWithScore = async (query, k, filter) => {
+            return await baseSimilaritySearchVectorWithScoreFn(query, k, filter ?? pgMetadataFilter)
+        }
+
+        const basePoolQueryFn = instance.pool.query.bind(instance.pool)
+
+        // @ts-ignore
+        instance.pool.query = async (queryString: string, parameters: any[]) => {
+            if (!instance.client) {
+                instance.client = await instance.pool.connect()
+            }
+
+            await instance.client.connect()
+
+            const whereClauseRegex = /WHERE ([^\n]+)/
+            let chatflowOr = ''
+
+            // Match chatflow uploaded file and keep filtering on other files:
+            // https://github.com/FlowiseAI/Flowise/pull/3367#discussion_r1804229295
+            if (chatId) {
+                parameters.push({ [FLOWISE_CHATID]: chatId })
+
+                chatflowOr = `OR metadata @> $${parameters.length}`
+            }
+
+            if (queryString.match(whereClauseRegex)) {
+                queryString = queryString.replace(whereClauseRegex, `WHERE (($1) AND NOT (metadata ? '${FLOWISE_CHATID}')) ${chatflowOr}`)
+            } else {
+                const orderByClauseRegex = /ORDER BY (.*)/
```

Comment:
is this order by the distanceStrategy?

Reply:
it's just a matcher to tweak original query in cas there is no where clause (no filter), i match the order by clause (which is always present) and insert the WHERE clause before.